### PR TITLE
[Doc] Add edge feature configurations doc

### PR DESCRIPTION
**Pull request by zhjwy9343**
*Issue #, if available:*

*Description of changes:*
This PR adds the two edge feature-related arguments to CLI configuration rst doc. And modifies the `node-feat-name` argument doc for consistency.

The preview page of the doc: https://james4graphstorm.readthedocs.io/en/james_ef_doc/cli/model-training-inference/configuration-run.html
 
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/source/cli/model-training-inference/configuration-run.rst [hunk 1]
```diff
@@ -70,16 +70,34 @@ GraphStorm provides a set of parameters to config the GNN model structure (input
     - Yaml: ``model_encoder_type: rgcn``
     - Argument: ``--model-encoder-type rgcn``
     - Default value: This parameter must be provided by user.
-- **node_feat_name**: User defined feature name. It accepts two formats: a) `fname`, if a node has node features, the corresponding feature name will be fname; b) `ntype0:feat0 ntype1:featA ...`, different node types have different node feature name(s). In the example, "ntype0" has a node feature named "feat0" and "ntype1" has a node feature named "featA".
+- **node_feat_name**: User defined feature name. It accepts two formats like: a) `fname`, if all node types have the same node feature name, the corresponding feature name will be `fname`; b) `ntype0:feat0 ntype1:featA ...`, different node types have different node feature name(s). In the below example, `ntype0` has a node feature named `feat0` and `ntype1` has two node features named `featA` and `featB`.
```
Comment (Oxfordblue7) [inline docs/source/cli/model-training-inference/configuration-run.rst:73]:
```suggestion
- **node_feat_name**: User defined feature name. It accepts two formats like: a) `fname`, if all node types have the same node feature name, the corresponding feature name will be `fname`; b) `ntype0:feat0 ntype1:featA ...`, if different node types have different node feature name(s). In the below example, `ntype0` has a node feature named `feat0` and `ntype1` has two node features named `featA` and `featB`.
```

File: docs/source/cli/model-training-inference/configuration-run.rst [hunk 2]
```diff
@@ -70,16 +70,34 @@ GraphStorm provides a set of parameters to config the GNN model structure (input
...
 
     - Yaml: ``node_feat_name:``
-                | ``- "ntype1:featA,featB"``
                 | ``- "ntype0:feat0"``
+                | ``- "ntype1:featA,featB"``
     - Argument: ``--node-feat-name "ntype0:feat0 ntype1:featA,featB"``
     - Default value: If not provided, there will be no node features used by GraphStorm even graphs have node features attached.
 
     .. Note:: Characters ``:`` and white space are not allowed to be used in node feature names.  And in Yaml format, need to put each node's feature in a separated line that starts with a hyphon.
```
Comment (Oxfordblue7) [inline docs/source/cli/model-training-inference/configuration-run.rst:81]:
```suggestion
    .. Note:: Characters ``:`` and white space are not allowed to be used in node feature names. In Yaml format, users need to put each node's feature in a separated line that starts with a hyphon.
```

File: docs/source/cli/model-training-inference/configuration-run.rst [hunk 3]
```diff
@@ -70,16 +70,34 @@ GraphStorm provides a set of parameters to config the GNN model structure (input
...
 
+- **edge_feat_name**: User defined edge feature name. It accepts two formats like: a) `fname`, if all edge types have the same feature name, the corresponding feature name will be `fname`; b) `src_ntype1,etype1,dst_ntype1:feat0,... src_ntype2,etype2,dst_ntype2:feata ...`, if different edge types have different feature name(s). In the below example, `src_ntype1,etype1,dst_ntype1` edge type has two features named `feat0`, and `feat1`, and `src_ntype2,etype2,dst_ntype2` edge type has one feature named `feata`.
```
Comment (Oxfordblue7) [inline docs/source/cli/model-training-inference/configuration-run.rst:83]:
```suggestion
- **edge_feat_name**: User defined edge feature name. It accepts two formats like: a) `fname`, if all edge types have the same feature name, the corresponding feature name will be `fname`; b) `src_ntype1,etype1,dst_ntype1:feat0,... src_ntype2,etype2,dst_ntype2:featA ...`, if different edge types have different feature name(s). In the below example, `src_ntype1,etype1,dst_ntype1` edge type has two edge features named `feat0`, and `feat1`, and `src_ntype2,etype2,dst_ntype2` edge type has one edge feature named `featA`.
```

File: docs/source/cli/model-training-inference/configuration-run.rst [hunk 4]
```diff
@@ -70,16 +70,34 @@ GraphStorm provides a set of parameters to config the GNN model structure (input
...
+
+    - Yaml: ``edge_feat_name:``
+                | ``- "src_ntype1,etype1,dst_ntype1:feat0,feat1"``
+                | ``- "src_ntype2,etype2,dst_ntype2:feata"``
```
Comment (Oxfordblue7) [inline docs/source/cli/model-training-inference/configuration-run.rst:87]:
```suggestion
                | ``- "src_ntype2,etype2,dst_ntype2:featA"``
```

File: docs/source/cli/model-training-inference/configuration-run.rst [hunk 5]
```diff
@@ -70,16 +70,34 @@ GraphStorm provides a set of parameters to config the GNN model structure (input
...
+    - Argument: ``—-edge-feat-name src_ntype1,etype1,dst_ntype1:feat0,feat1 src_ntype2,etype2dst_ntype2:feata``
```
Comment (Oxfordblue7) [inline docs/source/cli/model-training-inference/configuration-run.rst:88]:
```suggestion
    - Argument: ``—-edge-feat-name src_ntype1,etype1,dst_ntype1:feat0,feat1 src_ntype2,etype2dst_ntype2:featA``
```

File: docs/source/cli/model-training-inference/configuration-run.rst [hunk 6]
```diff
@@ -70,16 +70,34 @@ GraphStorm provides a set of parameters to config the GNN model structure (input
...
+    - Default value: If not provided, there will be no edge features used by GraphStorm even graphs have edge features attached.
+
+    .. Note:: In the latest version, the RGCN encoder has been modified to support using edge features during message passing computation. If users would like to use edge features, please set the ``model_encoder_type`` to be ``rgcn``.
```
Comment (Oxfordblue7) [inline docs/source/cli/model-training-inference/configuration-run.rst:91]: > In the latest version
Shall we specify the version?

Comment (Oxfordblue7) [inline docs/source/cli/model-training-inference/configuration-run.rst:91]: Shall we also explain what the outcomes will be if ``model_encoder_type`` is not set to ``rgcn`` when using edge features?

File: docs/source/cli/model-training-inference/configuration-run.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (zhjwy9343) [reply docs/source/cli/model-training-inference/configuration-run.rst:73]: Want to provide differences to let users know these names could be diverse.

File: docs/source/cli/model-training-inference/configuration-run.rst (hunk 6 above)
Comment (zhjwy9343) [reply docs/source/cli/model-training-inference/configuration-run.rst:91]: Agree. Changed to version 0.4

Comment (zhjwy9343) [reply docs/source/cli/model-training-inference/configuration-run.rst:91]: Updated with additional sentence, telling users there will be an assertion error.

**Review by Oxfordblue7 — APPROVED** [review]
LGTM.

**Pull request merged**